unittests/linuxliketests.py

```diff
@@ -1868,6 +1868,18 @@ def test_top_options_in_sp(self):
         testdir = os.path.join(self.unit_test_dir, '124 pkgsubproj')
         self.init(testdir)
 
+    def test_unreadable_dir_in_declare_dep(self):
+        testdir = os.path.join(self.unit_test_dir, '125 declare_dep var')
+        tmpdir = Path(tempfile.mkdtemp())
+        self.addCleanup(windows_proof_rmtree, tmpdir)
+        declaredepdir = tmpdir / 'test'
+        declaredepdir.mkdir()
+        try:
+            tmpdir.chmod(0o444)
+            self.init(testdir, extra_args=f'-Ddir={declaredepdir}')
+        finally:
+            tmpdir.chmod(0o755)
+
```

Comment:
This is essentially testing a design bug in the python standard library.

Reply:
To be clear, the purpose of my comment here is that I'm not convinced we need another unittest in order to check that this case doesn't regress.

Reply:
It's just an optimization... But since it was giving a PermissionError it was a buggy optimization which is why I added the test.